src/v2/providers/dataconnect.ts

```diff
@@ -0,0 +1,363 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import { CloudEvent, CloudFunction } from "../core";
+import { ParamsOf, VarName } from "../../common/params";
+import {
+  EventHandlerOptions,
+  getGlobalOptions,
+  optionsToEndpoint,
+  SupportedRegion,
+} from "../options";
+import { normalizePath } from "../../common/utilities/path";
+import { wrapTraceContext } from "../trace";
+import { withInit } from "../../common/onInit";
+import { initV2Endpoint, ManifestEndpoint } from "../../runtime/manifest";
+import { PathPattern } from "../../common/utilities/path-pattern";
+import { Expression } from "../../params";
+import { ResetValue } from "../../common/options";
+
+/** @internal */
+export const mutationExecutedEventType =
+  "google.firebase.dataconnect.connector.v1.mutationExecuted";
+
+/** @hidden */
+export interface SourceLocation {
+  line: number;
+  column: number;
+}
+
+/** @hidden */
+export interface GraphqlErrorExtensions {
+  file: string;
+  code: string;
+  debugDetails: string;
+}
+
+/** @hidden */
+export interface GraphqlError {
+  message: string;
+  locations: Array<SourceLocation>;
+  path: Array<string>;
+  extensions: GraphqlErrorExtensions;
+}
+
+/** @hidden */
+export interface RawMutation<V, R> {
+  data: R;
+  variables: V;
+  errors: Array<GraphqlError>;
+}
+
+/** @hidden */
+export interface MutationEventData<V, R> {
+  ["@type"]: "type.googleapis.com/google.events.firebase.dataconnect.v1.MutationEventData";
+  payload: RawMutation<V, R>;
+}
+
+/** @hidden */
+export interface RawDataConnectEvent<T> extends CloudEvent<T> {
+  project: string;
+  location: string;
+  service: string;
+  schema: string;
+  connector: string;
+  operation: string;
+  authtype: AuthType;
+  authid?: string;
+}
+
+/**
+ * AuthType defines the possible values for the authType field in a Firebase Data Connect event.
+ * - app_user: an end user of an application..
+ * - admin: an admin user of an application. In the context of impersonate endpoints used by the admin SDK, the impersonator.
+ * - unknown: a general type to capture all other principals not captured in the other auth types.
+ */
+export type AuthType = "app_user" | "admin" | "unknown";
+
+/** OperationOptions extend EventHandlerOptions with a provided service, connector, and operation. */
+export interface OperationOptions<
+  Service extends string = string,
+  Connector extends string = string,
+  Operation extends string = string
+> extends EventHandlerOptions {
+  /** Firebase Data Connect service ID */
+  service?: Service;
```

Comment:
i'm a bit surprised that all these options are optional 🤔  should the all be required? what does not providing one mean? is it interpreted as wildcard or something else